internal/controller/wrappers.go

```diff
@@ -52,6 +53,13 @@ func (d *DeploymentWrapper) GetMaxSurge() intstr.IntOrString {
 	return intstr.FromInt(0)
 }
 
+// AddAnnotation will reset and add new annotation map every time this func is called
+func (d *DeploymentWrapper) AddAnnotation(status, newReplicas string) {
+	//always need to new map to reset the status of deployment and clear previous ones
```

Comment:
this seems bad. you don't want to clear customer annotation with that make(map. you on;y want to clear your key/status.